jasper_reports/JasperReports/browse_data_generator.py

```diff
@@ -67,175 +68,170 @@ def warning(self, message):
+
+                    for rec_id in current_records:
+                        new = rec_id.copy()
+                        new[current_path] = v
+                        current_new_records.append(new)
+
                     relations2 = [f.partition('/')[2] for f in relations
                                   if f.partition('/')[0] == root and
                                   f.partition('/')[2]]
-                    newRecords += self.generateIds(v, relations2, currentPath,
-                                                   currentNewRecords)
+                    new_records += self.generate_ids(v, relations2,
+                                                     current_path,
+                                                     current_new_records)
 
-                currentRecords = newRecords
-        return currentRecords
+                current_records = new_records
+        return current_records
 
 
 class XmlBrowseDataGenerator(BrowseDataGenerator):
+    def __init__(self, report, model, env, cr, uid, ids, context):
+        super(XmlBrowseDataGenerator, self).__init__(report, model, env, cr,
+                                                     uid, ids, context)
+        self.all_records = []
+        self.document = None
+
     # XML file generation works as follows:
     # By default (if no OPENERP_RELATIONS property exists in the report)
     # a record will be created for each model id we've been asked to show.
     # If there are any elements in the OPENERP_RELATIONS list,
     # they will imply a LEFT JOIN like behaviour on the rows to be shown.
-    def generate(self, fileName):
-        self.allRecords = []
+    def generate(self, file_name):
+        self.all_records = []
         relations = self.report.relations()
-        # The following loop generates one entry to allRecords list
+        # The following loop generates one entry to all_records list
         # for each record that will be created. If there are any relations
         # it acts like a LEFT JOIN against the main model/table.
-        for record in self.env[self.model].browse(self.cr, self.uid,
-                                                  self.ids, self._context):
-            newRecords = self.generateIds(record, relations, '',
-                                          [{'root': record}])
+        for record in self.env[self.model].browse(self.ids):
+
+            new_records = self.generate_ids(record, relations, '',
+                                            [{'root': record}])
             copies = 1
-            if(self.report.copiesField() and record.__hasattr__
-               (self.report.copiesField())):
-                copies = int(record.__getattr__(self.report.copiesField()))
-            for new in newRecords:
+            if self.report.copiesField() and \
+                    record.__hasattr__(self.report.copies_field):
+                copies = int(record.__getattr__(self.report.copies_field))
```

Comment:
This code pass in pep8 check: https://travis-ci.org/multidadosti-erp/Jasperv8/jobs/188831847